js/src/Ice/ConnectionI.js

```diff
@@ -61,6 +58,16 @@ class MessageInfo {
     }
 }
 
+function scheduleCloseTimeout(connection) {
+    if (connection._closeTimeout > 0 && connection._closeTimeoutId === undefined) {
+        // Schedules a one-time check.
```

Comment:
So unlike other mappings, we schedule this close-timeout check only if no other close-timeout check is already scheduled. Whihc is fine.

Reply:
Unlike with other mappings, the event loop in JavaScript does not terminate while there are pending events. If multiple timers are scheduled, we need to maintain a list of them and cancel each one before the application exits; otherwise, the application may hang until all scheduled timers have fired.